docs/rfcs/0002_templates_mvp.md

```diff
@@ -0,0 +1,339 @@
+| fields_schema | JSON        | The template submission rules. A JSON blob with a standard structure which will encode the template field objects having things like: label, key, description, type, values_accepted, instructions |
+| created_at    | datetime    | Timestamp of creation                                                                                                                                                                              |
+| modified_at   | varchar     | Timestamp of update                                                                                                                                                                                |
+| creator       | foreign key | Foreign key to User that created the submission template                                                                                                                                           |
+
+#### Django Model Schema
+
+| Name                          | Type           | Description                                                                                            |
+| ----------------------------- | -------------- | ------------------------------------------------------------------------------------------------------ |
+| All fields in DB table schema |
+| released                      | BooleanField   | A computed field (not stored in db) denoting whether the template is being used in one or more studies |
+| submission_template_file      | FileField      | Reference to the actual template file in a storage system                                              |
+| extract_config_file           | FileField      | Reference to the actual extract config file in a storage system                                        |
+| studies                       | RelatedManager | List of studies this template version is assigned to                                                   |
+
+## ✨ Creating Templates
+
+### CreateTemplateMutation
+
+Creates a Template and initial TemplateVersion
+
+```graphql
+type CreateTemplateInput {
+    name: String!
+    owner: ID!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### CreateTemplateVersionMutation
+
+Creates a TemplateVersion and adds it to a Template
+
+```graphql
+type CreateTemplateVersionInput {
+    templateID: ID!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### (Optional) CloneTemplateMutation
+
+Creates a new Template and a TemplateVersion by duplicating the
+referenced template version and it's template. Overwrites the name,
+description, and owner of the source Template with the supplied values
+in the cloned Template.
+
+```graphql
+type CloneTemplateInput {
+    templateVersion: ID!
```

Comment:
What do you mean? I didn't include it in the mockups, but I'm assuming whatever UI we build for creating a new template version would allow you to create via copying an existing version.

Reply:
Are these versions found through browsing through all template versions in the current organization or is there some public template listing?

Reply:
The template list view in the PR description _is_ the public template listing. You can filter by study (and I think I forgot to add a filter for organization). Now that list view has rows that represent the `Template` not `TemplateVersion`. So if we added a copy/clone button the actions in the row, clicking it would copy the latest version of the `Template` and create a new `Template` and `TemplateVersion`. 

I haven't thought enough about the UI for copying a specific version of a template to create a new `Template` + `TemplateVersion`. But I imagine that could happen via the Edit Template View. Could add a copy button to the actions in the Versions list view, and then prompt the user to create a new TemplateVersion OR an entirely new template (Template + TemplateVersion).

Also, keep in mind that even if we didn't support creation through cloning, someone could always download a template (field defs, template file, etc) and manually create a new Template + TemplateVersion.